### PR TITLE
Add CI-Connect CLI

### PR DESCRIPTION
**Pull request by thschue**

### CODE REVIEW

**Review by eugenbelsky — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+
+  ############################################################################
+  # Build Container Images                                                   #
+  ############################################################################
+  container_build:
+    needs: [prepare_ci_run]
+    strategy:
+      matrix:
+        component: [ "ci-connect-cli" ]
```
Comment (didiladi) [inline .github/workflows/build.yaml:106]: Using a matrix with only one component is an overkill.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+        with:
+          context: ${{ matrix.component }}/.
+          push: ${{ github.ref == 'refs/heads/main' || github.ref == 'refs/heads/release-*' }}
+          file: ${{ matrix.component }}/docker/Dockerfile
+          platforms: linux/amd64
+          tags: |
+            ${{ env.DOCKER_ORGANIZATION }}/${{ matrix.component }}:latest-dev
+            ${{ env.DOCKER_ORGANIZATION }}/${{ matrix.component }}:${{ env.VERSION }}
+
+      - name: Run Trivy vulnerability scanner
```
Comment (didiladi) [inline .github/workflows/build.yaml:154]: :rocket:

File: ci-connect-cli/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Keptn CI Connect CLI
+
+Needed environment variables:
+```
+export GIT_REPO="https://github.com/<repo>"
```
Comment (didiladi) [inline ci-connect-cli/README.md:5]: This can also be a git repo not necessarily hosted by GitHub.

File: ci-connect-cli/cmd/fileutils.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+import (
+	"fmt"
+	"github.com/spf13/afero"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+/* MIT License
```
Comment (didiladi) [inline ci-connect-cli/cmd/fileutils.go:11]: This comment should be at the top of the file.

File: ci-connect-cli/cmd/gitutils.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
+package cmd
```
Comment (didiladi) [inline ci-connect-cli/cmd/gitutils.go:1]: The apache file header is missing from all the shource files.

File: ci-connect-cli/cmd/version.go [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+const version = "0.0.7"
```
Comment (didiladi) [inline ci-connect-cli/cmd/version.go:9]: Might make sense to increate the version to `0.1` or even `1.0.0`.

File: ci-connect-cli/go.mod [hunk 7]
```diff
@@ -0,0 +1,18 @@
+module github.com/keptn-sandbox/keptn-git-toolbox/ci-connect-cli
+
+go 1.16
+
+require (
+	github.com/go-git/go-git/v5 v5.4.2
```
Comment (didiladi) [inline ci-connect-cli/go.mod:6]: Without the patch of @yeahservice (`github.com/yeahservice/go-git/v5 v5.4.2-aws-patch`), using an AWS code commit repository for the keptn git repo will not be possible.

File: ci-connect-cli/main.go [hunk 8]
```diff
@@ -0,0 +1,22 @@
+/*
+Copyright © 2021 NAME HERE <EMAIL ADDRESS>
```
Comment (didiladi) [inline ci-connect-cli/main.go:2]: `2021 NAME HERE <EMAIL ADDRESS>`

File: ci-connect-cli/README.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+   user_name: jenkins
+```
+
+The deployment can be triggered using:
+
+```
+./ci-connect-cli trigger-deployment --service podtatoservice
+```
+
+Example configurations for testing can be found in the `.keptn` and `helm` directories
```
Comment (yeahservice) [inline ci-connect-cli/README.md:28]: Can remove this line since there are no configuration examples

**Review by yeahservice — CHANGES_REQUESTED** [review]

File: ci-connect-cli/helm/test-service/values.yaml [hunk 10]
```diff
@@ -0,0 +1 @@
+test
```
Comment (yeahservice) [inline ci-connect-cli/helm/test-service/values.yaml:1]: Can remove this file without the other testing files of `.keptn` it does nothing

**Review by yeahservice — CHANGES_REQUESTED** [review]

**Pull request merged**